### PR TITLE
feat: add ensureLink/ensureLinkSync for fs

### PR DESCRIPTION
**Pull request by axetroy**
part of #261.
ref: #268.

### CODE REVIEW

File: fs/ensure_link.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @param src the source file path. directory are not allow.
+ * @param dest the destination link path
+ */
+export async function ensureLink(src: string, dest: string): Promise<void> {
+  if (await exists(dest)) {
+    const destStatInfo = await Deno.lstat(dest);
+    const destFilePathType = getFileInfoType(destStatInfo);
+    if (destFilePathType !== PathType.file) {
+      throw new Error(
+        `Ensure path exists, expected 'file', got '${destFilePathType}'`
```
Comment (axetroy) [inline fs/ensure_link.ts]: Hard links don't have a specific type, so we can only use it as file.

File: fs/ensure_link.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import * as path from "./path/mod.ts";
+import { ensureDir, ensureDirSync } from "./ensure_dir.ts";
+import { exists, existsSync } from "./exists.ts";
+import { PathType, getFileInfoType } from "./utils.ts";
+
+/**
+ * Ensures that the hard link exists.
+ * If the directory structure does not exist, it is created.
+ *
+ * @param src the source file path. directory are not allow.
```
Comment (axetroy) [inline fs/ensure_link.ts]: hard link does not support for the directory.

Comment (ry) [reply fs/ensure_link.ts]: Can you add a TODO and/or issue for this?

Comment (axetroy) [reply fs/ensure_link.ts]: the system not support. This is not possible.

https://askubuntu.com/questions/210741/why-are-hard-links-not-allowed-for-directories

So I think todo is meaningless

Comment (ry) [reply fs/ensure_link.ts]: Ah ok. I didn’t know that.

Please change “directory are not allow” to “Directory hard links are not allowed”

Comment (ry) [reply fs/ensure_link.ts]: Ah ok. I didn’t know that.

Please change “directory are not allow” to “Directory hard links are not allowed”

**Pull request merged**